kornia/geometry/plane.py

```diff
@@ -3,104 +3,129 @@
 
 from typing import Optional
 
-from kornia.core import Module, Tensor, normalize
+from kornia.core import Module, Tensor, stack, where
+from kornia.core.tensor_wrapper import unwrap, wrap
 from kornia.geometry.linalg import batched_dot_product
-from kornia.testing import KORNIA_CHECK_SHAPE
+from kornia.geometry.vector import Scalar, Vector3
+from kornia.testing import KORNIA_CHECK, KORNIA_CHECK_SHAPE, KORNIA_CHECK_TYPE
 from kornia.utils.helpers import _torch_svd_cast
 
 __all__ = ["Hyperplane", "fit_plane"]
 
 
-# NOTE: in the near future the constructor will change
-# class Hyperplane(Module):
-#     def __init__(self, n: _VectorType, d: _VectorType) -> None:
-#         super().__init__()
-#         self._n = n
-#         self._d = d
+def normalized(v, eps=1e-6):
+    return v / batched_dot_product(v, v).add(eps).sqrt()
 
 
 class Hyperplane(Module):
-    def __init__(self, n: Tensor, d: Tensor) -> None:
+    def __init__(self, n: Vector3, d: Scalar) -> None:
```

Comment:
So the current HyperPlane will only for 3d case?

Reply:
we need to possibly add later again `VectorType` and subclass `Vector2` / `Vector3`